presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/reloading/ReloadingManagerSpec.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.resourceGroups.reloading;
+
+import com.facebook.presto.resourceGroups.ManagerSpec;
+import com.facebook.presto.resourceGroups.ResourceGroupIdTemplate;
+import com.facebook.presto.resourceGroups.ResourceGroupSpec;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+public class ReloadingManagerSpec
```

Comment:
We have an objected called ManagerSpec already, and reloading is the name given to it because the ReloadingManger essentially just reloads all the new information provided to it through a new ManagerSpec object provided by an implementation of the ManagerSpecProvider interface.